tests/__snapshots__/index.test.ts.snap

```diff
@@ -1231,7 +1242,10 @@ export default class TypeEnum extends React.Component<Props> {
       foo: _pt.string.isRequired
     }), _pt.shape({
       bar: _pt.number
-    })])
+    })]),
+    typeRefEnumIntersection: _pt.oneOfType([_pt.oneOf([Baz.Bar]), _pt.oneOf([Qux.Bar])]),
+    typeRefEnumUnion: _pt.oneOfType([_pt.oneOf([Baz.Foo]), _pt.oneOf([Baz.Bar]), _pt.oneOf([Qux.Bar])]),
+    typeRefEnum: _pt.oneOf([Qux.Foo, Qux.Bar])
```

Comment:
This should be `typeRefEnum: _pt.oneOf([0, 1])`, since we want JS values, not TS ones, and since enums are index based by default.

Reply:
Should we?
Enums are available in the runtime like objects. 
In enum's members can be a lot of expressions and calculations/computed values.
And a lot of cases where user can try to use some runtime variable - some constant like value for it.

But I've made some changes(locally) what you have noticed, but I'm not sure that is good idea.

Reply:
Yeah but this example is referencing the *values* of the enum, which is the value of the `Foo` (0) and `Bar` (1) members. Prop types operate on JS literal values, so an enum object doesn't make much sense.